tests/components/vesync/test_switch.py

```diff
@@ -0,0 +1,27 @@
+"""Tests for the switch module."""
+from syrupy import SnapshotAssertion
+
+from homeassistant.core import HomeAssistant
+
+from .common import get_entities, get_states
+
+
+async def test_async_setup_entry__loaded_switches(
+    hass: HomeAssistant,
+    setup_platform,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test the resulting setup state is as expected for the platform."""
+
+    states = {}
+    identifier = "outlet"
+    entities = get_entities(hass, identifier)
+    assert len(entities) == 7
+    states[identifier] = get_states(hass, entities)
+
+    identifier = "switch"
+    entities = get_entities(hass, identifier)
+    assert len(entities) == 1
+    states[identifier] = get_states(hass, entities)
+
+    assert states == snapshot
```

Comment:
Ok, so if I am reading your suggestion correctly, your issue appears to be with the way I was telling it which devices I wanted to check the state of, instead of just walking ALL entities and ensuring state is as expected.

So, in order to actually implement your suggestion and have it be unique for each platform, I need to completely change how the input fixture files are constructed.  The current set of files submits all test devices to the platform for configuration - always.  To be clear, if I put your code suggestion in each platform test file, it'll result in the same exact output each time.  I'll need some time think this through, but I think I need to combine your idea for the override_platforms() fixture , but instead apply it to the requests_mock fixture so that it injects only the devices needed for testing.

Thanks for your feedback, I'll get this thing straightened out soon!

Reply:
Thanks for the example code.  After reviewing, I can see that while my original code matched your intention (using device registry, entity registry, and state), it failed to do both positive AND negative testing.